src/daft-parquet/src/stream_reader.rs

```diff
@@ -40,6 +49,109 @@ fn prune_fields_from_schema(
     }
 }
 
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn local_parquet_read_into_column_iters(
+    uri: &str,
+    columns: Option<&[String]>,
+    start_offset: Option<usize>,
+    num_rows: Option<usize>,
+    row_groups: Option<&[i64]>,
+    predicate: Option<ExprRef>,
+    schema_infer_options: ParquetSchemaInferenceOptions,
+    metadata: Option<Arc<parquet2::metadata::FileMetaData>>,
+    chunk_size: usize,
+) -> super::Result<(
+    Arc<parquet2::metadata::FileMetaData>,
+    SchemaRef,
+    Vec<RowGroupRange>,
+    impl Iterator<Item = super::Result<ArrowChunkIters>>,
+)> {
+    const LOCAL_PROTOCOL: &str = "file://";
+    let uri = uri
+        .strip_prefix(LOCAL_PROTOCOL)
+        .map(|s| s.to_string())
+        .unwrap_or(uri.to_string());
+
+    let mut reader = File::open(uri.clone()).with_context(|_| super::InternalIOSnafu {
+        path: uri.to_string(),
+    })?;
+    let size = reader
+        .metadata()
+        .with_context(|_| super::InternalIOSnafu {
+            path: uri.to_string(),
+        })?
+        .len();
+
+    if size < 12 {
+        return Err(super::Error::FileTooSmall {
+            path: uri,
+            file_size: size as usize,
+        });
+    }
+
+    // Use block in place to read metadata as the current function is in an asynchronous context.
+    let metadata = match metadata {
+        Some(m) => m,
+        None => read::read_metadata(&mut reader)
```

Comment:
This is a blocking call in an asynchronous context. However since it is a metadata read, not sure if it's worth the overhead of calling it in a blocking thread i.e. spawn_blocking / rayon.